### PR TITLE
Initial Merge of the crunchy-postgres-ha & crunchy-postgres-gis-ha Containers

### PR DESCRIPTION
**Pull request by andrewlecuyer**
This PR represents the initial merge of the `crunchy-postgres-ha` and `crunchy-postgres-gis-ha` containers into the Crunchy Container Suite.  These containers provide a new means for creating highly-available PostgreSQL clusters in a container environment, specifically leveraging Patroni to bootstrap and then manage PG clusters.  This includes directly leveraging the distributed configuration store provided by Kubernetes to effectively bootstrap PG clusters (i.e. by means of a leader election process), and then ensure those clusters remain highly available (i.e. by providing extensive failover capabilities).

By default, the container is able to set logical defaults for required PG and Patroni configuration in order to simplify highly-available PG cluster deployments in a container environment.  However, the container is meant to be fully customizable and configurable.  This means any auto-configuration or default configuration options/settings can be disabled and/or overridden, while additional configuration parameters can be provided as needed.  This includes the ability to provide custom Patroni and PostgreSQL configuration files.

The container supports workflows for both initializing new clusters, as well as converting existing PostgreSQL databases into Patroni clusters.  The containers also enable pgBackRest by default, while also making pgBackRest the default tool utilized for creating replicas.  And similar to the `crunchy-postgres` and `crunch-postgres-gis` containers, custom SQL can be run during initialization, while various modules (e.g. pgMonitor) can also be loaded when initializing a cluster.

Included in the PR are Dockerfiles for both CentOS and RHEL builds of the `crunchy-postgres-ha` and `crunchy-postgres-gis-ha` containers, as well as any scripts and configuration files needed by those containers.  Examples are also included for running both the `crunchy-postgres-ha` and `crunchy-postgres-gis-ha` containers (these examples currently only support `ReadWriteMany` volumes pending the migration of the dedicated backrest repository container from the Postgres Operator project to the Crunchy Container Suite).

This PR also introduces the `crunchy-admin` container, which is a new administrative container that is designed to be run as a sidecar to the `crunchy-postgres-ha` and `crunchy-postgres-gis-ha` containers.  Configured with an account called `crunchyadm`, this container is able to authenticate into the database as this user using peer authentication, specifically by mounting a shared volume with the database container that contains a PG unix socket.

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

**What is the new behavior (if this is a feature change)?**

The `crunchy-ha` and `crunchy-postgres-ha-gis` containers can now be utilized 

**Other information**:

### CODE REVIEW

File: bin/postgres-ha/kube-ssh-wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+source "/tmp/pod_env.sh"
+
+CLUSTER_LABEL="pg-cluster=${CLUSTER_NAME}"
+TARGET_LABEL="pgo-backrest-repo=true"
+
+opts=$(echo "$@" | grep -o "\-\-c.*")
+pod=$(kubectl get pods --selector=${CLUSTER_LABEL},${TARGET_LABEL} -o name)
+
+exec kubectl exec -i "${pod}" -- bash -c "pgbackrest ${opts}"
```
Comment (jkatz) [inline bin/postgres-ha/kube-ssh-wrapper.sh:11]: @andrewlecuyer Note that we may need the `oc` equivalent here -- not for this commit.

File: bin/postgres-ha/post-existing-init.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CREATE USER ${PATRONI_SUPERUSER_USERNAME} WITH SUPERUSER ENCRYPTED PASSWORD '${PATRONI_SUPERUSER_PASSWORD}';
```
Comment (jkatz) [inline bin/postgres-ha/post-existing-init.sql]: @andrewlecuyer The `ENCRYPTED` keyword does not exist in newer versions of PostgreSQL. I can't remember when it was deprecrated, but that keyword can be removed.

File: bin/pull-from-crunchy.sh [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 REG_CCP_IMAGE_PREFIX=registry.crunchydata.com/crunchydata
-for CONTAINER in crunchy-prometheus crunchy-upgrade crunchy-grafana crunchy-collect crunchy-pgbadger crunchy-pgpool crunchy-backup crunchy-postgres crunchy-postgres-gis crunchy-pgbouncer crunchy-pgadmin4 crunchy-pgdump crunchy-pgrestore crunchy-backrest-restore crunchy-scheduler crunchy-pgbench crunchy-pgbasebackup-restore
+for CONTAINER in crunchy-prometheus crunchy-upgrade crunchy-grafana crunchy-collect crunchy-pgbadger crunchy-pgpool crunchy-backup crunchy-postgres crunchy-postgres-gis crunchy-pgbouncer crunchy-pgadmin4 crunchy-pgdump crunchy-pgrestore crunchy-backrest-restore crunchy-scheduler crunchy-pgbench crunchy-pgbasebackup-restore crunchy-postgres-ha
```
Comment (jkatz) [inline bin/pull-from-crunchy.sh]: @andrewlecuyer `crunchy-postgres-gis-ha` too?

File: bin/pull-from-dockerhub.sh [hunk 4]
```diff
@@ -41,3 +41,4 @@ docker pull $CCP_IMAGE_PREFIX/crunchy-upgrade:$CCP_IMAGE_TAG
 docker pull $CCP_IMAGE_PREFIX/crunchy-scheduler:$CCP_IMAGE_TAG
 docker pull $CCP_IMAGE_PREFIX/crunchy-pgbench:$CCP_IMAGE_TAG
 docker pull $CCP_IMAGE_PREFIX/crunchy-pgbasebackup-restore:$CCP_IMAGE_TAG
+docker pull $CCP_IMAGE_PREFIX/crunchy-postgres-ha:$CCP_IMAGE_TAG
```
Comment (jkatz) [inline bin/pull-from-dockerhub.sh:44]: @andrewlecuyer `crunchy-postgres-gis-ha` too?

File: bin/pull-from-gcr.sh [hunk 5]
```diff
@@ -21,7 +21,8 @@ IMAGES=(
     crunchy-prometheus
     crunchy-scheduler
     crunchy-upgrade
-    crunchy-pgbasebackup-restore
+    crunchy-pgbasebackup-restore.
+    crunchy-postgres-ha
```
Comment (jkatz) [inline bin/pull-from-gcr.sh:25]: @andrewlecuyer `crunchy-postgres-gis-ha` too?

File: bin/pull-from-gcr.sh [hunk 6]
```diff
@@ -21,7 +21,8 @@ IMAGES=(
     crunchy-prometheus
     crunchy-scheduler
     crunchy-upgrade
-    crunchy-pgbasebackup-restore
+    crunchy-pgbasebackup-restore.
```
Comment (jkatz) [inline bin/pull-from-gcr.sh]: @andrewlecuyer There's an extra `.` after `crunchy-pgbasebackup-restore`

File: bin/push-to-dockerhub.sh [hunk 7]
```diff
@@ -28,3 +28,4 @@ docker push  $CCP_IMAGE_PREFIX/crunchy-scheduler:$CCP_IMAGE_TAG
 docker push $CCP_IMAGE_PREFIX/crunchy-upgrade:$CCP_IMAGE_TAG
 docker push $CCP_IMAGE_PREFIX/crunchy-pgbench:$CCP_IMAGE_TAG
 docker push $CCP_IMAGE_PREFIX/crunchy-pgbasebackup-restore:$CCP_IMAGE_TAG
+docker push  $CCP_IMAGE_PREFIX/crunchy-postgres-ha:$CCP_IMAGE_TAG
```
Comment (jkatz) [inline bin/push-to-dockerhub.sh:31]: @andrewlecuyer crunchy-postgres-gis-ha too?

File: bin/push-to-local.sh [hunk 8]
```diff
@@ -21,6 +21,7 @@ IMAGES=(
     crunchy-scheduler
     crunchy-upgrade
     crunchy-backrest-restore
+    crunchy-postgres-ha
```
Comment (jkatz) [inline bin/push-to-local.sh:24]: @andrewlecuyer crunchy-postgres-gis-ha too?

File: bin/remove-all-images.sh [hunk 9]
```diff
@@ -16,7 +16,7 @@
 for i in \
 restore pgdump pgrestore postgres-gis prometheus grafana collect pgbadger pgpool \
 watch backup postgres pgbouncer pgadmin4 scheduler upgrade backrest-restore pgbench \
-pgbasebackup-restore
+pgbasebackup-restore postgres-ha
```
Comment (jkatz) [inline bin/remove-all-images.sh]: @andrewlecuyer crunchy-postgres-gis-ha too?

**Review by jkatz — CHANGES_REQUESTED** [review]
Ensure the crunchy-postgres-gis-ha containers as well as admin container are available in the Makefile, push, pull etc. files.

File: bin/pull-from-crunchy.sh (hunk 3 above)
Comment (andrewlecuyer) [reply bin/pull-from-crunchy.sh]: `crunchy-postgres-gis-ha` is now included

File: bin/pull-from-dockerhub.sh (hunk 4 above)
Comment (andrewlecuyer) [reply bin/pull-from-dockerhub.sh:44]: `crunchy-postgres-gis-ha` is now included

File: bin/pull-from-gcr.sh (hunk 5 above)
Comment (andrewlecuyer) [reply bin/pull-from-gcr.sh:25]: `crunchy-postgres-gis-ha` is now included

File: bin/pull-from-gcr.sh (hunk 6 above)
Comment (andrewlecuyer) [reply bin/pull-from-gcr.sh]: Extra `.` removed

File: bin/push-to-dockerhub.sh (hunk 7 above)
Comment (andrewlecuyer) [reply bin/push-to-dockerhub.sh:31]: `crunchy-postgres-gis-ha` is now included

File: bin/push-to-local.sh (hunk 8 above)
Comment (andrewlecuyer) [reply bin/push-to-local.sh:24]: `crunchy-postgres-gis-ha` is now included

File: bin/remove-all-images.sh (hunk 9 above)
Comment (andrewlecuyer) [reply bin/remove-all-images.sh]: `crunchy-postgres-gis-ha` is now included

File: bin/postgres-ha/post-existing-init.sql (hunk 2 above)
Comment (andrewlecuyer) [reply bin/postgres-ha/post-existing-init.sql]: The `ENCRYPTED` parameter has been removed.

**Review by jkatz — APPROVED** [review]

**Pull request merged**